validation/jinja2/validation/_macros.html

```diff
@@ -57,30 +23,33 @@
  # Adapted from: https://docs.djangoproject.com/en/2.1/topics/pagination/
  #}
 {% macro pagination_with_query(page, endpoint, query, encode_query_with_page) %}
-<ul class="nav justify-content-center">
-  {% if page.has_previous() %}
-  <li class="nav-item"><a class="nav__link" href={{ encode_query_with_page(query, page=1) }}>
-      First
-  </a></li>
-  <li class="nav-item"><a class="nav__link" href={{ encode_query_with_page(query, page=page.previous_page_number()) }}>
-      Previous
-  </a></li>
-  {% endif %}
-
-  <li class="nav-item"><a class="nav__button active" data-cy="nav-button-with-query" href="#">
-      Page {{ page.number }} of {{ page.paginator.num_pages }}
-  </a></li>
-
-  {% if page.has_next() %}
-  <li class="nav-item"><a class="nav__link" href={{ encode_query_with_page(query, page=page.next_page_number()) }}>
-      Next
-  </a></li>
-  <li class="nav-item"><a class="nav__link" href={{ encode_query_with_page(query, page=page.paginator.num_pages) }}>
-      Last
-  </a></li>
-  {% endif %}
-
-</ul>
+    <ul class="nav justify-content-center">
+        {% if page.has_previous() %}
+            <li class="nav-item"><a class="nav__link" data-cy="nav-first" href={{ encode_query_with_page(query, page=1) }}>
+                First
+            </a></li>
+            <li class="nav-item"><a class="nav__link" data-cy="nav-prev" href={{ encode_query_with_page(query, page=page.previous_page_number()) }}>
+                Previous
+            </a></li>
+        {% endif %}
+
+        <li class="nav-item"><a class="nav__button active" data-cy="nav-button-with-query" href="#">
+            Page {{ page.number }} of {{ page.paginator.num_pages }}
+        </a></li>
+
+        {% if page.has_next() %}
+            <li class="nav-item"><a class="nav__link" data-cy="nav-next" href={{ encode_query_with_page(query, page=page.next_page_number()) }}>
+                Next
+            </a></li>
+            <li class="nav-item"><a class="nav__link" data-cy="nav-last" href={{ encode_query_with_page(query, page=page.paginator.num_pages) }}>
+                Last
+            </a></li>
+        {% endif %}
+
+    </ul>
+    <div class="center">
+        Jump to Page: <input type="text" data-cy="page-num-input" class="input--short"> <button type="button" data-cy="go-button-nav" class="button button--success button--small">Go</button>
```

Comment:
I'm not 100% sure this has the same intended effect but this could be replaced by a form (zero JavaScript):

```suggestion
        <form method="GET">
            <label> Jump to Page: <input type="number" name="page" data-cy="page-num-input" class="input--short"></label>
            <button type="submit" data-cy="go-button-nav" class="button button--success button--small">Go</button>
        </form>
```

Reply:
I tried using a form initially and couldn't quite figure it out. Consider that `page=` often isn't the only parameter. I would also likely pull the form out of `macros` since it needs to `csrf_token`. I'm open to it if it works though!

Reply:
Oh yeah, you're right! This will destroy any and all other params other than page! Note, `<form method="GET">` does not requrire a CSRF token; [only `POST` (and other, hipster methods) require the CSRF token ](https://docs.djangoproject.com/en/3.2/ref/csrf/#how-to-use-it).

Okay, keep the JS! 😂

Reply:
*do not actually do the following*, I only bring it up because a puzzle was implicitly posed:

Make it work by adding `<input type="hidden">` on the server side for every existing GET param?